easybuild/tools/github.py

```diff
@@ -716,6 +736,74 @@ def _easyconfigs_pr_common(paths, start_branch=None, pr_branch=None, target_acco
     return file_info, deleted_paths, git_repo, pr_branch, diff_stat
 
 
+def det_patch_specs(patch_paths, file_info):
+    """ Determine software names for patch files """
+    print_msg("determining software names for patch files...")
+    patch_specs = []
+    for patch_path in patch_paths:
+        soft_name = None
+        patch_file = os.path.basename(patch_path)
+
+        # consider patch lists of easyconfigs being provided
+        for ec in file_info['ecs']:
+            if patch_file in ec['patches']:
+                soft_name = ec.name
+                break
+
+        if soft_name:
+            patch_specs.append((patch_path, soft_name))
+        else:
+            # fall back on scanning all eb files for patches
+            print "Matching easyconfig for %s not found on the first try:" % patch_path,
+            print "scanning all easyconfigs to determine where patch file belongs (this may take a while)..."
+            soft_name = scan_all_easyconfigs(patch_file)
+            if soft_name:
+                patch_specs.append((patch_path, soft_name))
+            else:
+                # still nothing found
+                raise EasyBuildError("Failed to determine software name to which patch file %s relates", patch_path)
+
+    return patch_specs
+
+
+def scan_all_easyconfigs(patch_name):
+    """
+    Scan all easyconfigs in the robot path to determine which software a patch file belongs to
+
+    @param patch_name: name of the patch file
+    """
+    robot_paths = build_option('robot_path')
+    soft_name = ""
+    found = False
+    for robot_path in robot_paths:
+        all_ecs = []
```

Comment:
move this out of this loop, and construct a *full* set of easyconfigs first before iterating over them?

```python
all_ecs = []
for robot_path in robot_paths:
    ...

for idx, filename in enumerate(all_ecs):
    ...